packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -1209,6 +1214,23 @@ export default {
     updateFilters(updatedFilters) {
       this.$store.commit(SET_SAVED_FILTERS, updatedFilters);
     },
+
+    analysisFindingSelection(findingObject) {
+      const finding = findingObject.resolvedFinding && findingObject.resolvedFinding.finding;
+      if (!finding) return;
+
+      if (finding.checkId === 'n-plus-one-query') {
+        this.$store.commit(SET_HIGHLIGHTED_EVENTS, finding.relatedEvents);
+        this.$store.commit(SET_VIEW, VIEW_FLAMEGRAPH);
+
+        const eventToFocus =
+          finding.participatingEvents && finding.participatingEvents.commonAncestor;
```

Comment:
The nullish operator works in this project now.

I'd also fall back to the finding event if `commonAncestor` isn't present.

```
const eventToFocus = finding.participatingEvents?.commonAncestor || finding.event;
```

Reply:
Good idea :+1: